initializr-generator/src/main/java/io/spring/initializr/generator/project/ProjectDescriptionModifications.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2012-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.spring.initializr.generator.project;
+
+import io.spring.initializr.generator.buildsystem.BuildSystem;
+import io.spring.initializr.generator.version.Version;
+import java.util.Objects;
+import java.util.function.BiConsumer;
+
+public class ProjectDescriptionModifications {
+
+	private MutableProjectDescription original;
+	private MutableProjectDescription copy;
+
+	public ProjectDescriptionModifications(final MutableProjectDescription original) {
+		
+		final MutableProjectDescription copy = new MutableProjectDescription();
```

Comment:
That doesn't work if one extends `ProjectDescription` to make it a custom one. That's the exact reason why I used a copy constructor as it forces a sub-class to override it. There is actually such example in the doc (and in integration tests) ans we need to support the case a user provides its own view of `ProjectDescription`.

Reply:
Agreed - I planned on moving into copy constructor.